src/fsipy/simulations/predeform.py

```diff
@@ -0,0 +1,210 @@
+            checkpoint_step=50,  # Save frequency of checkpoint files
+            # Linear solver parameters
+            linear_solver="mumps",
+            atol=1e-6,  # Absolute tolerance in the Newton solver
+            rtol=1e-6,  # Relative tolerance in the Newton solver
+            recompute=20,  # Recompute the Jacobian matix within time steps
+            recompute_tstep=20,  # Recompute the Jacobian matix over time steps
+            # boundary condition parameters
+            mesh_path="mesh/cylinder.h5",
+            inlet_id=2,  # inlet id for the fluid
+            inlet_outlet_s_id=11,  # inlet and outlet id for solid
+            fsi_id=22,  # id for fsi interface
+            rigid_id=11,  # "rigid wall" id for the fluid and mesh problem
+            outer_wall_id=33,  # id for the outer surface of the solid
+            # Fluid parameters
+            rho_f=1.025e3,  # Fluid density [kg/m3]
+            mu_f=3.5e-3,  # Fluid dynamic viscosity [Pa.s]
+            dx_f_id=1,  # ID of marker in the fluid domain. When reading the mesh, the fuid domain is assigned with a 1.
+            # Pre-deform parameters
+            v_max_final=0.75,  # Final max centerline velocity of parabolic profile
+            # should be the cycle-averaged average velocity for your main simulation
+            P_final=10000,  # Steady State pressure applied to wall
+            # should be your cycle-averaged gage pressure for your main simulation
+            t_start_v=0.0,  # Start time for ramping up velocity
+            t_end_v=0.1,  # End time for ramping up velocity
+            t_start_p=0.1,  # Start time for ramping up pressure
+            t_end_p=0.2,  # End time for ramping up pressure
+            # Solid parameters
+            rho_s=1.0e3,  # Solid density [kg/m3]
+            mu_s=mu_s_val,  # Solid shear modulus or 2nd Lame Coef. [Pa]
+            nu_s=nu_s_val,  # Solid Poisson ratio [-]
+            lambda_s=lambda_s_val,  # Solid 1st Lame Coef. [Pa]
+            dx_s_id=2,  # ID of marker in the solid domain
+            # mesh lifting parameters (see turtleFSI for options)
+            extrapolation="laplace",  # laplace, elastic, biharmonic, no-extrapolation
+            extrapolation_sub_type="constant",  # ["constant","small_constant","volume","volume_change","bc1","bc2"]
+            folder="predeform_results",  # output folder generated for simulation
+            save_deg=1,  # save_deg=1 saves corner nodes only, save_deg=2 saves corner + mid-point nodes for viz
+        )
+    )
+
+    return default_variables
+
+
+def get_mesh_domain_and_boundaries(mesh_path, folder, **namespace):
+    print("Obtaining mesh, domains and boundaries...")
+    mesh = Mesh()
+    hdf = HDF5File(mesh.mpi_comm(), mesh_path, "r")
+    hdf.read(mesh, "/mesh", False)
+    boundaries = MeshFunction("size_t", mesh, 2)
+    hdf.read(boundaries, "/boundaries")
+    domains = MeshFunction("size_t", mesh, 3)
+    hdf.read(domains, "/domains")
+
```

Comment:
In case of aneurysm, we would want to modify fsi region like we do in `aneurysm.py`. I think it’s worth adding that in here.